cqrs-codegen/src/lib.rs

```diff
@@ -52,6 +52,50 @@ pub fn derive_event(input: TokenStream) -> TokenStream {
     expand(input, event::derive)
 }
 
+/// Derives [`cqrs::RegisteredEvent`] implementation for structs and enums.
+///
+/// # Structs
+///
+/// When deriving [`cqrs::RegisteredEvent`] for struct, the struct is treated as
+/// a single distinct event.
+///
+/// # Enums
+///
+/// When deriving [`cqrs::RegisteredEvent`] for enum, the enum is treated as
+/// a sum-type representing a set of possible events.
+///
+/// In practice this means, that [`cqrs::RegisteredEvent`] can only be derived
+/// for an enum when all variants of such enum have exactly one field (variant
+/// can be either a tuple-variant or a struct-variant).
+///
+/// Generated implementation of [`cqrs::RegisteredEvent::type_id`] would
+/// match on all variants and return [`core::any::TypeId`] of each variant's
+/// field (__without proxying call to it__, so fields don't have to implement
+/// [`cqrs::RegisteredEvent`] themself).
```

Comment:
Do not forget to update docs. Never.

Reply:
My bad. I've actually thought about it... and then I didn't. :/